pootle/apps/pootle_store/views.py

```diff
@@ -584,8 +584,22 @@ def submit(request, unit):
     form = form_class(request.POST, instance=unit, request=request)
 
     if form.is_valid():
+        suggestion = None
+        suggid = form.cleaned_data['suggestion_id']
```

Comment:
hi @ta2-1 - im wondering if this logic should be in the form.save method - although i would keep the request handling - ie raise Http404 in the view